packages/pyright-internal/src/analyzer/service.ts

```diff
@@ -527,33 +526,43 @@ export class AnalyzerService {
     // an optional config file, and default values.
     private _getConfigOptions(host: Host, commandLineOptions: CommandLineOptions): ConfigOptions {
-        let projectRoot = realCasePath(commandLineOptions.executionRoot, this.fs);
-        let configFilePath: string | undefined;
-        let pyprojectFilePath: string | undefined;
+        let projectRoot = this.fs.realCasePath(
+            Uri.file(commandLineOptions.executionRoot, this._isFileSystemCaseSensitve, /* checkRelative */ true)
+        );
+        const executionRoot = this.fs.realCasePath(
+            Uri.file(commandLineOptions.executionRoot, this._isFileSystemCaseSensitve, /* checkRelative */ true)
+        );
+        let configFilePath: Uri | undefined;
+        let pyprojectFilePath: Uri | undefined;
 
         if (commandLineOptions.configFilePath) {
             // If the config file path was specified, determine whether it's
             // a directory (in which case the default config file name is assumed)
             // or a file.
-            configFilePath = realCasePath(
-                combinePaths(commandLineOptions.executionRoot, normalizePath(commandLineOptions.configFilePath)),
-                this.fs
+            configFilePath = this.fs.realCasePath(
+                isRootedDiskPath(commandLineOptions.configFilePath)
+                    ? Uri.file(
+                          commandLineOptions.configFilePath,
+                          this._isFileSystemCaseSensitve,
+                          /* checkRelative */ true
+                      )
+                    : projectRoot.combinePaths(commandLineOptions.configFilePath)
             );
             if (!this.fs.existsSync(configFilePath)) {
-                this._console.info(`Configuration file not found at ${configFilePath}.`);
-                configFilePath = realCasePath(commandLineOptions.executionRoot, this.fs);
+                this._console.info(`Configuration file not found at ${configFilePath.toUserVisibleString()}.`);
+                configFilePath = projectRoot;
             } else {
-                if (configFilePath.toLowerCase().endsWith('.json')) {
-                    projectRoot = realCasePath(getDirectoryPath(configFilePath), this.fs);
+                if (configFilePath.extension.endsWith('.json')) {
+                    projectRoot = configFilePath.getDirectory();
                 } else {
                     projectRoot = configFilePath;
                     configFilePath = this._findConfigFile(configFilePath);
                     if (!configFilePath) {
-                        this._console.info(`Configuration file not found at ${projectRoot}.`);
+                        this._console.info(`Configuration file not found at ${projectRoot.toUserVisibleString()}.`);
                     }
                 }
             }
-        } else if (projectRoot) {
+        } else if (this.fs.existsSync(projectRoot)) {
```

Comment:
Yeah this seems wrong. It should check if the executionRoot is empty, not if the project root exists.

Reply:
The old code would create the project root from the realcasepath of the executionRoot. If this was the empty string it would just return the empty string and this if case would be false. To have the same logic I should check the executionRoot.